### PR TITLE
XCTestCase Extensions 

### PR DESCRIPTION
**Pull request by concertman**
Here's a quick, unformatted dump of the items that @chris59david and I were using on another project.

### CODE REVIEW

File: XCTestAdditions/XCTestAdditions.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -35,6 +37,8 @@
 /* End PBXContainerItemProxy section */
 
 /* Begin PBXFileReference section */
+		7AD98E5C1EC3264C0067FB53 /* XCTestCaseExtentions.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = XCTestCaseExtentions.swift; sourceTree = "<group>"; };
```
Comment (concertman) [inline XCTestAdditions/XCTestAdditions.xcodeproj/project.pbxproj]: Why would we want to use ObjC style naming when we are writing Swift? ;)

File: XCTestAdditions/XCTestAdditionsUITests/XCElementExtentions.swift [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+//    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+//    SOFTWARE.
+
+import Foundation
+import XCTest
+
```
Comment (madhikarma) [inline XCTestAdditions/XCTestAdditionsUITests/XCElementExtentions.swift]: Could we add a one liner documentation for what the extension is for? 

I'm also wondering if we need to rename to 

`XCUIElement+TextInput.swift`

I'm just wary of calling things +Additions or +Extensions

File: XCTestAdditions/XCTestAdditionsUITests/XCTestCaseExtentions.swift [hunk 3]
```diff
@@ -0,0 +1,87 @@
+//    The MIT License (MIT)
+//
+//    Copyright (c) 2016 ustwo™
```
Comment (madhikarma) [inline XCTestAdditions/XCTestAdditionsUITests/XCTestCaseExtentions.swift]: 2017

File: XCTestAdditions/XCTestAdditionsUITests/XCTestCaseExtentions.swift [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+//    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+//    SOFTWARE.
+
+import Foundation
+import XCTest
+
+extension XCTestCase {
+
```
Comment (madhikarma) [inline XCTestAdditions/XCTestAdditionsUITests/XCTestCaseExtentions.swift]: Documentation? Or we can do it later

File: XCTestAdditions/XCTestAdditionsUITests/XCTestCaseExtentions.swift [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+//    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+//    SOFTWARE.
+
+import Foundation
+import XCTest
+
```
Comment (madhikarma) [inline XCTestAdditions/XCTestAdditionsUITests/XCTestCaseExtentions.swift]: Documentation? Or we can do it later

File: XCTestAdditions/XCTestAdditionsUITests/XCTestCaseExtentions.swift [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+        expectation(for: existsPredicate, evaluatedWith: XCElement, handler: nil)
+        
+        waitForExpectations(timeout: timeOut) { (error) in
+            if error != nil {
+                self.recordFailure(withDescription: errorMessage, inFile: fileName, atLine: lineNumber, expected: true)
+            }
+        }
+    }
+}
+
```
Comment (madhikarma) [inline XCTestAdditions/XCTestAdditionsUITests/XCTestCaseExtentions.swift]: Should we move into it's own file?

**Comment by madhikarma** [comment]
Merging now and will open a new PR for some of the updates mentioned above (if agreed upon)

**Pull request merged**